src/plugins/resolve-media-queries-plugin.js

```diff
@@ -73,7 +74,7 @@ function _topLevelRulesToCSS({
     );
 
     // CSS classes cannot start with a number
-    const mediaQueryClassName = 'rmq-' + hash(query + ruleCSS);
+    const mediaQueryClassName = `rmq-${styleId()}-${hash(query + ruleCSS)}`;
```

Comment:
Sure looks that way. The unique component id route may solve that issue, and I think it'd be a similar sized diff. Thoughts?